tests/src/Gui/Camera.cpp

```diff
@@ -0,0 +1,220 @@
+{
+    // Set up a simple view volume to test the projection of the unit vectors.
+    // The actual values don't matter much, as we are only interested in the
+    // relative lengths of the projected vectors.
+    SbViewVolume volume;
+    // left, right, bottom, top, near, far
+    volume.ortho(-10, 10, -10, 10, -10, 10);
+
+    volume.rotateCamera(rot);
+    const auto matrix = volume.getMatrix();
+
+    // Get the transformed unit vectors
+    SbVec3f vo, vx, vy, vz;
+    matrix.multVecMatrix(SbVec3f(0, 0, 0), vo);
+    matrix.multVecMatrix(SbVec3f(10, 0, 0), vx);
+    matrix.multVecMatrix(SbVec3f(0, 10, 0), vy);
+    matrix.multVecMatrix(SbVec3f(0, 0, 10), vz);
+
+    // Project to XY plane by setting Z to 0
+    vo[2] = 0;
+    vx[2] = 0;
+    vy[2] = 0;
+    vz[2] = 0;
+
+    // Return the lengths of the projected vectors
+    return {(vx - vo).length(), (vy - vo).length(), (vz - vo).length()};
+}
+
+}  // namespace
+
+TEST(CameraPrecalculatedQuaternions, testIsometric)
+{
+    // Use the formula to get the isometric rotation
+    double alpha = toRadians(45.0f);
+    double beta = std::asin(-std::sqrt(1.0 / 3.0));
+
+    const Rotation actual = buildAxonometricRotation(alpha, beta);
+    const Rotation expected = toRotation(Gui::Camera::isometric());
+
+    EXPECT_TRUE(actual.isSame(expected, 1e-6));
+}
+
+TEST(CameraPrecalculatedQuaternions, testDimetric)
+{
+    // Use the formula to get the dimetric rotation
+    double alpha = std::asin(std::sqrt(1.0 / 8.0));
+    double beta = -std::asin(1.0 / 3.0);
+
+    const Rotation actual = buildAxonometricRotation(alpha, beta);
+    const Rotation expected = toRotation(Gui::Camera::dimetric());
+
+    EXPECT_TRUE(actual.isSame(expected, 1e-6));
+}
+
+TEST(CameraPrecalculatedQuaternions, testTrimetric)
+{
+    // Use the formula to get the trimetric rotation
+    double alpha = toRadians(30.0);
+    double beta = toRadians(-35.0);
+
+    const Rotation actual = buildAxonometricRotation(alpha, beta);
+    const Rotation expected = toRotation(Gui::Camera::trimetric());
+
+    EXPECT_TRUE(actual.isSame(expected, 1e-6));
+}
+
+TEST(CameraRotation, testIsometricProjection)
+{
+    auto rot = Gui::Camera::isometric();
+    auto lengths = getProjectedLengths(rot);
+
+    // In isometric, expect all lengths to be roughly equal
+    EXPECT_NEAR(lengths[0], lengths[1], 1e-6);  // X == Y
+    EXPECT_NEAR(lengths[0], lengths[2], 1e-6);  // X == Z
+    EXPECT_NEAR(lengths[1], lengths[2], 1e-6);  // Y == Z
+}
+
+TEST(CameraRotation, testDimetricProjection)
+{
+    auto rot = Gui::Camera::dimetric();
+    auto lengths = getProjectedLengths(rot);
+
+    // In dimetric, expect two lengths to be roughly equal, one different
+    int similarCount = 0;
+    if (std::abs(lengths[0] - lengths[1]) < 1e-6) {
+        similarCount++;
+    }
+    if (std::abs(lengths[1] - lengths[2]) < 1e-6) {
+        similarCount++;
+    }
+    if (std::abs(lengths[0] - lengths[2]) < 1e-6) {
+        similarCount++;
+    }
```

Comment:
1. I think that `1e-6` should really be `constexpr double tolerance = 1e-6`. It is magic number and repeated.
2. I'd consider using `std::count_if` in that case:
```cpp

// the type is std::initializer_list<std::pair<double, double>> - feel free to put it explicitly, I feel that it is too verbose
constexpr auto pairs = {
    { lengths[0], lengths[1] },
    { lengths[1], lengths[2] },
    { lengths[0], lengths[2] },
}

const auto isSimilar = [](std::pair<double, double> lengths) -> bool {
    return std::abs(lengths.first - lengths.second) < tolerance;
}

unsigned similarCount = std::ranges::count_if(pairs, isSimilar)
```

I'm writing it from github editor so probably has some errors - idea should stay however.

Reply:
You are correct, that would be nicer. This is a test though so I lowered the standard a bit, but I agree and will fix it, thanks!